config/webpack/webpack.config.js

```diff
@@ -70,6 +71,13 @@ module.exports = function (dirname, config = {}) {
             new CircularDependencyPlugin({
                 exclude: /(node_modules|examples)\/./,
                 failOnError: false // https://github.com/nodejs/readable-stream/issues/280#issuecomment-297076462
+            }),
+            new GlobalizePlugin({
+                production: false,
+                developmentLocale: "en",
+                supportedLocales: ["en", "es", "fr"],
+                messages: "../../src/i18n/[locale].json",
```

Comment:
I added the path in the variable name for each variable as you describe like:
{
    "en": {
        "application": {
            "edit": "Edit"
        },
        "editor": {
            "label": "Code Editor"
        }
    }
}
The only problem remaining is really to store a partial version of the file under each folder and be able to load those messages.
There is Globalize.loadmessages(), but if I can find how to load it from this method, it should work
I will look again how we can load separate files for each module, but If you think of something, let me know.
Thanks